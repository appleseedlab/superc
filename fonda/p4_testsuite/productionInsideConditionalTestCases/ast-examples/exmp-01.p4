header int_data_t {
    #ifdef CONFIG_A
        bit<4> flag;
    #else
        bit<4> flag;
    #endif
}
