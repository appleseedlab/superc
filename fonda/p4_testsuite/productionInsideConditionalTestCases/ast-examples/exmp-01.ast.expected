TranslationUnit(
  start(
    program(
      input(
        Conditional(
          ,
          declaration(
            typeDeclaration(
              derivedTypeDeclaration(
                headerTypeDeclaration(
                  optAnnotations(),
                  superc.core.Syntax$Language(
                    "header"
                  ),
                  name(
                    nonTypeName(
                      superc.core.Syntax$Text(
                        "int_data_t"
                      )
                    )
                  ),
                  optTypeParameters(),
                  superc.core.Syntax$Language(
                    "{"
                  ),
                  structFieldList(
                    Conditional(
                      ,
                      Conditional(
                        (declare-fun |(defined CONFIG_A)| () Bool)(assert |(defined CONFIG_A)|),
                        structField(
                          optAnnotations(),
                          typeRef(
                            baseType(
                              superc.core.Syntax$Language(
                                "bit"
                              ),
                              l_angle(
                                superc.core.Syntax$Language(
                                  "<"
                                )
                              ),
                              superc.core.Syntax$Text(
                                "4"
                              ),
                              r_angle(
                                superc.core.Syntax$Language(
                                  ">"
                                )
                              )
                            )
                          ),
                          name(
                            nonTypeName(
                              superc.core.Syntax$Text(
                                "flag"
                              )
                            )
                          ),
                          superc.core.Syntax$Language(
                            ";"
                          )
                        ),
                        (declare-fun |(defined CONFIG_A)| () Bool)(assert (not |(defined CONFIG_A)|)),
                        structField(
                          optAnnotations(),
                          typeRef(
                            baseType(
                              superc.core.Syntax$Language(
                                "bit"
                              ),
                              l_angle(
                                superc.core.Syntax$Language(
                                  "<"
                                )
                              ),
                              superc.core.Syntax$Text(
                                "4"
                              ),
                              r_angle(
                                superc.core.Syntax$Language(
                                  ">"
                                )
                              )
                            )
                          ),
                          name(
                            nonTypeName(
                              superc.core.Syntax$Text(
                                "flag"
                              )
                            )
                          ),
                          superc.core.Syntax$Language(
                            ";"
                          )
                        )
                      )
                    )
                  ),
                  superc.core.Syntax$Language(
                    "}"
                  )
                )
              )
            )
          )
        )
      )
    )
  )
)
